### PR TITLE
refactor: un-nest requireValidTransition

### PR DESCRIPTION
**Pull request by geoknee**
Preparatory work for tackling #3035 

- It is now a very thin wrapper around two pure functions, avoiding the need for cross-contract calls 
- The branching logic in the wrapper should be simple to emulate in other languages and _is_ emulated in the `server-wallet` - The pure functions can be emulated separately. My strategy here is to use PureEVM for the app check, and write a typescript function that manually emulates the protocol check. The justification for this being that the protocol check is unlikely to change frequently and also easy to make an exact match for. 
- the typescript function in question has been placed in `nitro-protocol` and has an accompanying test suite. 
- this is imported into `server-wallet`, where it replaces the existing function. 
- side effects of this: I had to move the special logic around setup states into the existing `shouldValidateTransition` function. 

This is good because no such special case exists on chain and I want to keep the transition validation as close to the solidity code as possible. However, this also means that *no* checks at all are performed in the setup phase (so the app definition could change, for example). I think this may be an acceptable regression for now.

### CODE REVIEW

File: packages/nitro-protocol/contracts/ForceMove.sol [hunk 1]
```diff
@@ -547,25 +547,25 @@ contract ForceMove is IForceMove {
         return stateHashes;
     }
 
+    enum IsValidTransition {True, NeedToCheckApp}
```
Comment (lalexgap) [inline packages/nitro-protocol/contracts/ForceMove.sol:550]: nit: `TransitionIsValid` (or something) might be a clearer name for the `True` enum?

**Review by lalexgap — APPROVED** [review]
This looks great to me and I'd be happy to get it merged in!!

Like you mentioned in standup we should probably think about when we validate a state vs when we store a state, but that can probably take place in a separate scope of work (after we spec it out in a notion doc)?

File: packages/nitro-protocol/contracts/ForceMove.sol [hunk 2]
```diff
@@ -547,25 +547,25 @@ contract ForceMove is IForceMove {
...
     /**
     * @notice Check that the submitted pair of states form a valid transition
     * @dev Check that the submitted pair of states form a valid transition
     * @param nParticipants Number of participants in the channel.
     transition
     * @param isFinalAB Pair of booleans denoting whether the first and second state (resp.) are final.
     * @param ab Variable parts of each of the pair of states
-    * @param turnNumB turnNum of the later state of the pair.
-    * @param appDefinition Address of deployed contract containing application-specific validTransition function.
+    * @param turnNumB turnNum of the later state of the pai
```
Comment (geoknee) [inline packages/nitro-protocol/contracts/ForceMove.sol:559]:
```suggestion
    * @param turnNumB turnNum of the later state of the pair
```

File: packages/nitro-protocol/src/valid-transition.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+export enum Status {
+  True,
+  NeedToCheckApp,
+}
+
+/**
+ *  Port of ForceMove#_requireValidProtocolTransition solidity code
+ * either:
+ * - returns Status.True (no further checks required)
+ * -returns  Status.NeedToCheckApp (clears protocol checks, overall validity requires App.validTransition)
```
Comment (geoknee) [inline packages/nitro-protocol/src/valid-transition.ts:14]:
```suggestion
 * - returns  Status.NeedToCheckApp (clears protocol checks, overall validity requires App.validTransition)
```

**Pull request merged**